dotnet/src/InternalUtilities/src/Diagnostics/ActivityExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Diagnostics;
+
+[ExcludeFromCodeCoverage]
+internal static class ActivityExtensions
+{
+    public static Activity? StartActivityWithTags(this ActivitySource source, string name, List<KeyValuePair<string, object?>> tags)
+    {
+        return source.StartActivity(
+            name,
+            ActivityKind.Internal,
+            Activity.Current?.Context ?? new ActivityContext(),
+            tags);
+    }
+
+    public static Activity AddTags(this Activity activity, List<KeyValuePair<string, object?>> tags)
+    {
+        tags.ForEach(tag =>
+        {
+            activity.SetTag(tag.Key, tag.Value);
+        });
+
+        return activity;
+    }
+
+    public static Activity AttachSensitiveDataAsEvent(this Activity activity, string name, List<KeyValuePair<string, object?>> tags)
+    {
+        activity.AddEvent(new ActivityEvent(
```

Comment:
I've been trying to make it work with EventSource or ILogger or OTel Logs and it's just impossible. So for .NET span events is the only choice for the time being 😓